### PR TITLE
Adding the scale_x_date function to the ucrAxes 

### LINKED ISSUE #5266: Add scale_x/y_dates option to Plot Options sub dialog
Here is a useful graph that is really simple to do, with a line plot
image

I would now like to scale the x-axis differently, giving a label every (say) 5 years, rather than every 20 years.
In the sub-dialogue, we therefore need to be able to specify the limits and the increment for dates.  Currently, the x-axis has 3 options, i.e. Continuous, Discrete, and Date.  But the Date options have yet to be completed

a) Please could they be, even if initially only partially.  And initially perhaps for the x-axis, though it should be the same for x and y.

In the meantime, I assume I could alter the script file.  I assume that is an easy thing to add?

And, by the way, in the Rwanda workshop we were asked the same question for the inventory plot.  It will greatly enhance the inventory plots to have access to the sub-dialogues.

### PR DESCRIPTION
**Pull request by Ivanluv**
fixes #5266

### CODE REVIEW

**Comment by rdstern** [comment]
Great so far - I am impressed and delighted.  I tested with Dodoma with temperaure data against date.
1) It does not keep the setting.  When you return to the sub-dialogue it has returned to Continuous.
2) The option to change the breaks works - great.  But it needs another argument to control the formatting.  So,  (date_breaks = "5 years", date_labels = "%Y").  Currently you include the date_breaks and you need to be able to add the date_labels as well.  @dannyparsons says this is probably the same control to add as in the Make Date dialogue.  It is a pull-down with the common options, but where you can also type your own, because there are so many.
3) The option Limits doesn't quite work.  If you change the lower limit it doesn't give the upper limit and then says the second argument is missing.  If you then also change the upper limit it says the first argument is missing.

**Comment by Ivanluv** [comment]
@rdstern  am still working on the issue of the  subdialogue returning to continuous

**Comment by Ivanluv** [comment]
Apart from the issue of the subdialogue returnig to the continous could @Shadrack  review this so that it can be merged and I write up an issue on the pending changes

**Comment by lloyddewit** [comment]
@Ivanluv I saw that you did a commit since @rdstern 's request for change, but wasn't sure if you still planned additional changes. Is this PR now ready for retest/review?

**Comment by Ivanluv** [comment]
@lloyddewit  its not ready for review yet

**Comment by lloyddewit** [comment]
@Ivanluv I saw you made a new commit. Is it ready for review?

**Comment by Ivanluv** [comment]
Yes,this is ready for review.However @rdstern  I have not implemented the suggestion of finding the maximum and minimum dates.I can open a new issue on that and have this reviewed and merged

**Comment by lloyddewit** [comment]
@rdstern Please could you approve again so we can merge? thanks
I also noticed that you documented a couple of remaining issues in issue #5823 .

**Comment by rdstern** [comment]
@lloyddewit I have now approved again.  I have also suggested that @Patowhiz may be able to resolve the 2 points quite quickly in the new issue, #5823 .

File: instat/ucrAxes.vb [hunk 1]
```diff
@@ -257,6 +334,14 @@ Public Class ucrAxes
...
         End If
 
+        If clsXYScaleDateFunction.ContainsParameter("date_breaks") Then
+            clsXYScaleDateBreakOperator = clsXYScaleDateFunction.GetParameter("date_breaks").clsArgumentCodeStructure
+        End If
+
+        If clsXYScaleDateFunction.ContainsParameter("limits") Then
+            clsXYScaleDateLimitFunction = clsXYScaleDateFunction.GetParameter("limits").clsArgumentCodeStructure
+        End If
+
```
Comment (shadrackkibet) [inline instat/ucrAxes.vb:344]: @Ivanluv @rdstern @lloyddewit these changes is causing issues on graphics dialogs. When you click on **Plot Options ** button in any of the graphics dialogs the software breaks here and an exception is thrown as shown below. 

image

**Comment by lloyddewit** [comment]
@shadrackkibet  Thank you for spotting this. I was able to trigger the same exception by clicking:
- Histogram plot diagram `Plot Options` button
- Two variable graph `Options` button

This exception is probably also triggered in other dialog boxes.

@Ivanluv Please could you try to recreate and investigate this issue? Please could you document your findings in a new GitHub issue and then try and resolve? thanks

**Pull request merged**